### PR TITLE
Limit comments

### LINKED ISSUE #35: Do not comment if too many issues found
Include a threshold that if total issue count is over a certain amount it will not comment per line anymore. It will still post a commit status.

Things to consider:

1. What should the threshold be, 32? (Allow user to override it from command line)
2. Do we want to not comment at all or still place the first X comments?

### PR DESCRIPTION
**Pull request by balloob**
This will limit the total number of comments that Farcy will add to a single PR, taking existing comments by Farcy into consideration. Currently defaulted to 128.

Includes an extra commit because branch is based on PR #39.

This PR fixes #35.

### CODE REVIEW

File: farcy/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """Farcy, a code review bot for github pull requests.
 
-Usage: farcy.py [-D | --logging=LEVEL]
+Usage: farcy.py [-D | --logging=LEVEL] [--comments-per-pr=LIMIT]
```
Comment (bboe) [inline farcy/__init__.py]: This should probably be on its own line to be easier to read.

File: farcy/__init__.py [hunk 2]
```diff
@@ -39,7 +40,8 @@
     COMMIT_STATUS_FORMAT, FARCY_COMMENT_START, CONFIG_DIR)
 from .exceptions import FarcyException, HandlerException
 from .helpers import (
-    added_lines, UTC, issues_by_line, split_dict, subtract_issues_by_line)
+    added_lines, filter_comments_from_farcy, issues_by_line, split_dict,
```
Comment (bboe) [inline farcy/__init__.py]: Technically UTC should be first `U` < `a`. I don't care that much though.

File: farcy/__init__.py [hunk 3]
```diff
@@ -246,7 +250,9 @@ def handle_pr(self, pr):
         issue_count = 0
         did_work = True
         exception = False
-        existing_comments = pr.review_comments()
+        existing_comments = list(filter_comments_from_farcy(
```
Comment (bboe) [inline farcy/__init__.py]: Does this exist in the helper? (Maybe it wasn't added?)

File: farcy/__init__.py [hunk 4]
```diff
@@ -302,6 +308,9 @@ def handle_pr(self, pr):
                           pr.number, file_issue_count,
                           's' if file_issue_count > 1 else '', pfile.filename))
 
+            if comments_added > self.pr_issue_report_limit:
```
Comment (bboe) [inline farcy/__init__.py]: Should this be `>=`?

File: farcy/__init__.py [hunk 5]
```diff
@@ -326,6 +335,9 @@ def handle_pr(self, pr):
                     info = pr.create_review_comment(*args).html_url['href']
                 self.log.info('PR#{0} ({1}:{2}) COMMENT: "{3}"'.format(
                     pr.number, pfile.filename, lineno, info))
+                comments_added += 1
```
Comment (bboe) [inline farcy/__init__.py]: Why is it that the previous conditional does `continue` and this one does `break`? Should only the former one do `break`, or should we always find out how many issues there are?

File: farcy/__init__.py (hunk 4 above)
Comment (balloob) [inline farcy/__init__.py]: Good catch.

File: farcy/__init__.py (hunk 2 above)
Comment (balloob) [inline farcy/__init__.py]: 78be62f68a84116500dd40052c820fbed5ab2831658d6cd3c0836c110d35a88c

Comment (bboe) [inline farcy/__init__.py]: Hey now! :-D

File: farcy/__init__.py (hunk 3 above)
Comment (balloob) [inline farcy/__init__.py]: It was a [building block](https://github.com/appfolio/farcy/blob/master/farcy/helpers.py#L40) of one of the other helpers. [It is even tested.](https://github.com/appfolio/farcy/blob/master/test/test_helpers.py#L31)

Comment (bboe) [inline farcy/__init__.py]: Doh! Somehow I missed it when looking at helpers.py.

**Comment by balloob** [comment]
Can we merge it?

**Comment by bboe** [comment]
Yes.

**Pull request merged**